app/controllers/notifications_controller.rb

```diff
@@ -2,11 +2,37 @@ class NotificationsController < AuthenticatedController
   include ProjectScoped
 
   def index
-    @notifications = current_user.notifications.newest.limit(20)
-    @unread_count = @notifications.unread.count
-
+    # TODO this doesn't fix all the N+1 query problems, needs a closer look:
+    notifs = current_user.notifications.newest.includes(:actor, notifiable: [:user, :commentable])
     respond_to do |format|
-      format.js
+      format.html do
+        @notifications = notifs.page(params[:page])
+      end
+      format.js do
+        @notifications = notifs.limit(20)
+        # NB the unread count is not the same as @notifications.count because
+        # @notifications a) includes read notifs and b) is capped at 20
+        @unread_count  = current_user.notifications.unread.count
+      end
+    end
+  end
+
+  def read
```

Comment:
would it be possible to use `update` method here?
passing the `id` or a `all` as a parameter.

If we ever want to mix api and web controllers (not sure about it), it looks more restful.